tests/ManagerTestCase.php

```diff
@@ -164,14 +164,30 @@ public function testDropDatabases()
     
     public function testConnectionInformationDecoded()
     {
+      $conn = null;
+      $thrownException = null;
       $dsn = 'mysql://' . urlencode('test/t') . ':' . urlencode('p@ssword') . '@localhost/' . urlencode('db/name');
 
-      $conn = Doctrine_Manager::connection($dsn);
-      $options = $conn->getOptions();
+      try {
+          $conn = Doctrine_Manager::connection($dsn);
+          $options = $conn->getOptions();
 
-      $this->assertEqual($options['username'], 'test/t');
-      $this->assertEqual($options['password'], 'p@ssword');
-      $this->assertEqual($options['dsn'], 'mysql:host=localhost;dbname=db/name');
+          $this->assertEqual($options['username'], 'test/t');
+          $this->assertEqual($options['password'], 'p@ssword');
+          $this->assertEqual($options['dsn'], 'mysql:host=localhost;dbname=db/name');
+      } catch (Throwable $e) {
+          $thrownException = $e;
+      } catch (Exception $e) {
+          $thrownException = $e;
```

Comment:
``
```suggestion
          // Exception catching is needed for PHP v5.x, as for PHP 7.x and 8.x we catch the \Throwable earlier
          $thrownException = $e;
```

Reply:
Nice catch of this bad code structure.

But instead of adding comment, I will fix the code structure.